sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -583,11 +583,12 @@ def _schedule_periodic_shutdown(self):
     # type: () -> None
     def shutdown_inactive_bundle_processors():
       # type: () -> None
-      for descriptor_id, last_access_time in self.last_access_times.items():
-        if (time.time() - last_access_time >
-            DEFAULT_BUNDLE_PROCESSOR_CACHE_SHUTDOWN_THRESHOLD_S):
-          BundleProcessorCache._shutdown_cached_bundle_processors(
-              self.cached_bundle_processors[descriptor_id])
+      with self._lock:
```

Comment:
from the issue, I see this is hard to reproduce but asking anyway, did you reproduce the error by any chance? If yes, can we add a test?

Reply:
@AnandInguva I was not sure how to reproduce it. I can take a look based on @tvalentyn comments and try reproducing it locally. This is my first contribution to the project so any guidance is appreciated. 

@tvalentyn Do you mean moving the lock definition before `self._schedule_periodic_shutdown()` ? i.e. swap lines 441 and 442?

Reply:
@lrakla ah sorry. I wasn't aware this is your first contribution. I am not sure on how to reproduce the error yet but I can try it on my local and let you know of any results and the right place to add test(if we could reproduce it).

Reply:
Thanks, @AnandInguva . @lrakla yes, that what I meant.

Reply:
@AnandInguva I will try it out too, and make the change @tvalentyn suggested.

Reply:
I tried few simple examples with 1000, 10000, 100000 elements by setting `direct_num_worker=4` and `direct_running_mode=multi_threading`. I haven't been able to reproduced the error.

Reply:
tried it with 1. I will try with 0.1